ir/memory.cpp

```diff
@@ -195,6 +195,125 @@ ostream& operator<<(ostream &os, const Pointer &p) {
 }
 
 
+Byte::Byte(const Memory &m, expr &&byterepr): m(m), p(move(byterepr)) {
+  assert(p.bits() == m.bitsByte());
+}
+
+Byte::Byte(const Pointer &ptr, unsigned i, expr non_poison)
+    : m(ptr.getMemory()) {
+  // TODO: support pointers larger than 64 bits.
+  assert(m.bitsMemorySize() <= 64 && m.bitsMemorySize() % 8 == 0);
+  // non_poison should be either bool or 1 bit.
+  if (non_poison.isBool())
+    non_poison = non_poison.toBVBool();
+  assert(non_poison.bits() == 1);
```

Comment:
this assert is redundant

Reply:
This is helpful when reader doesn't know what bitsByte() is, but I don't stick to it, so removed it